### ISSUE TITLE
[java] UnusedPrivateMethod for method referenced by lombok.Builder.ObtainVia

### ISSUE THREAD

**Issue by kdebski85** [pmd/pmd#5110]
**Affects PMD Version:** 
7.3.0

**Rule:**

[UnusedPrivateMethod](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:**

**Code Sample demonstrating the issue:**

```java
import lombok.Builder;
import lombok.Builder.ObtainVia;

import java.util.Collections;
import java.util.List;

@Builder(toBuilder = true)
public class ObtainViaTest {

    @ObtainVia(method = "fooProvider")
    private List<String> foo;

    private List<String> fooProvider() {
        return Collections.emptyList();
    }
}
```

**Expected outcome:**

PMD reports a violation at line 14, but that's wrong. That's a false positive.

**Running PMD through:** *Gradle*

**Comment by kdebski85** [comment]
Added PR https://github.com/pmd/pmd/pull/5111 to fix it
